snuba/query/snql/parser.py

```diff
@@ -871,6 +872,54 @@ def mangle_column_value(exp: Expression) -> Expression:
         query.transform_expressions(mangle_column_value)
 
 
+def _check_conditions_for_columns(
+    key: EntityKey, condition: Optional[Expression], table_name: str = "",
+) -> None:
+    entity = get_entity(key)
+    required = entity.get_required_conditions()
+    if not required:
+        return
+    elif condition is None:
+        raise ParsingException(
+            f"{key} requires conditions on {', '.join([r for r in required])}"
+        )
+    else:
+        conditions = get_first_level_and_conditions(condition)
+        seen = {r: False for r in required}
+        for condition in conditions:
+            if isinstance(condition, FunctionCall) and condition.parameters:
+                param = condition.parameters[0]
+                if isinstance(param, Column) and seen.get(param.column_name) == False:
+                    if not table_name or param.table_name == table_name:
+                        allowed_ops = required[param.column_name]
+                        if condition.function_name in allowed_ops:
+                            seen[param.column_name] = True
```

Comment:
Have you consider expressing the mandatory condition as matcher? That would give the flexibility of imposing conditions on the right hand side and you would not need custom code to validate each condition.